src/main/java/org/truffleruby/language/LazyRubyNode.java

```diff
@@ -87,4 +92,14 @@ private RubyNode getOrCreateMasterResolution() {
 
         return NodeUtil.cloneNode(masterResolution);
     }
+
+    @Override
+    public SourceSection getSourceSection() {
+        if (sourceSection == null) {
+            sourceSection = sourceSectionSupplier.get();
+        }
+
+        return sourceSection;
+    }
```

Comment:
We used to `resolve()` on getSourceSection(), but I removed it in 21e8b2c22bee1acac1fc72ded37441c3c50362b0, thinking notifyInserted() was enough.
But it might not be enough.
Not sure whether `resolve()` or this is better.

Reply:
The advantage of `resolve()` would be to also see sub-nodes and their SourceSection, not just this specific node's SourceSection.

Reply:
Yes `resolve` was the right thing to use - I've updated.